src/devices/bus/hp9825_optroms/hp9825_optrom.cpp

```diff
@@ -0,0 +1,184 @@
+// license:BSD-3-Clause
+// copyright-holders: F. Ulivi
+/*********************************************************************
+
+    hp9825_optrom.cpp
+
+    Optional ROMs for HP9825 systems
+
+*********************************************************************/
+
+#include "emu.h"
+#include "hp9825_optrom.h"
+#include "softlist.h"
+
+// Debugging
+//#define VERBOSE 1
+#include "logmacro.h"
+
+DEFINE_DEVICE_TYPE(HP9825_OPTROM_CART, hp9825_optrom_cart_device, "hp9825_optrom_cart", "HP9825 optional ROM cartridge")
+DEFINE_DEVICE_TYPE(HP9825_OPTROM_SLOT, hp9825_optrom_slot_device, "hp9825_optrom_slot", "HP9825 optional ROM slot")
+
+struct optrom_region_t {
+	offs_t m_start;
+	offs_t m_size;
+	const char *m_tag;
+};
```

Comment:
POSIX reserves all names ending in `_t` at global scope.  I realise MAME already violates this with `device_t` and a few other things, but we'd rather reduce the number of them than increase them.